### PR TITLE
[n/a] move user config merge to module, simplify TW service

### PR DESCRIPTION
**Pull request by gregkohn**
- Moves all of the config defaults & merging with user settings to the Module
- Reworks the `Tailwind` service to load the JSON when it is initialized, renames some methods

### CODE REVIEW

File: src/services/PartsKit.php [hunk 1]
```diff
@@ -33,19 +35,7 @@ public static function isRequest(): bool
...
-                'directory' => 'parts-kit',
-                'layout' => '_layouts/app',
-                'volume' => 'partsKit',
-                'theme' => 'light',
-            ],
-            $userSettings
-        );
-
-        return $config[$key] ?? null;
+        return Module::$config['partsKit'][$key] ?? null;
```
Comment (gregkohn) [inline src/services/PartsKit.php:38]: so for a moment i completely dropped this method and replaced the instances of `self::getConfig` with `Module::$config['partsKit']` -- buuuut we are actually using this method in the twig template to see which layout to extend. figured it was fine to keep, not sure if there's a better way to expose the static var to twig 🤷

**Review by joshuapease — APPROVED** [review]

File: src/services/PartsKit.php (hunk 1 above)
Comment (davist11) [reply src/services/PartsKit.php:38]: Yeah agree, makes sense to keep

File: src/services/Tailwind.php [hunk 2]
```diff
@@ -3,67 +3,44 @@
...
-        );
-
-        return $config[$key] ?? null;
+        parent::init();
+        $this->_loadFullConfig();
     }
 
     /**
      * Get Tailwind config object
      */
```
Comment (davist11) [inline src/services/Tailwind.php:28]: Minor but can you update these docblocks with the return types

File: src/services/Tailwind.php [hunk 3]
```diff
@@ -3,67 +3,44 @@
...
      * Get classname-ready list of colors
      */
     public function getColors(): array
     {
-        $config = $this->getTwConfig();
-        if (!$config) return [];
+        if (!$this->tailwindConfig) return [];
 
-        $colors = $config->theme->colors;
+        $colors = $this->tailwindConfig['theme']['colors'];
```
Comment (davist11) [inline src/services/Tailwind.php]: Worth `?? null` ?

File: tests/_craft/config/tailwind/tailwind.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+                "400": "#cbd5e0",
+                "500": "#a0aec0",
+                "600": "#718096",
+                "700": "#4a5568",
+                "800": "#2d3748",
+                "900": "#1a202c"
+            }
+        }
+    }
+}
```
Comment (davist11) [inline tests/_craft/config/tailwind/tailwind.json:21]: This feels so much nicer being a JSON file

**Review by davist11 — APPROVED** [review]
Looks great, couple of minor formatting things

**Pull request merged**